packages/driver/src/util/tab_activation.ts

```diff
@@ -0,0 +1,43 @@
+import type { ICypress } from '../cypress'
+
+function activateMainTab () {
+  // Don't need to activate the main tab if it already has focus
+  if (document.hasFocus()) return
+
+  return new Promise<void>((resolve) => {
+    const url = `${window.location.origin}${window.location.pathname}`
+
+    // This sends a message on the window that the extension content script
+    // listens for in order to carry out activating the main tab
+    window.postMessage({ message: 'extension:activate:main:tab', url }, '*')
```

Comment:
Do we want to scope these messages to Cypress in anyway?

```suggestion
    window.postMessage({ message: 'cypress:extension:activate:main:tab', url }, '*')
```

Reply:
Addressed in https://github.com/cypress-io/cypress/pull/28334/commits/48468486ba9da3321128140ec9109a4b6f0f1844